### ISSUE TITLE
Tempoup and then tempodown leads to skipping of video frames

### ISSUE THREAD

**Issue by antonsoroko** [xbmc/xbmc#25784]
### Describe the bug
Here is a clear and concise description of what the problem is:

If you do tempoup and then tempodown to normal speed - Kodi skips video frames (or moves to next keyframe maybe). Anyway it looks like it skips from 0.5 to 3 second of video.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

If you do tempoup and then tempodown to normal speed - nothing is skipped.

## Actual Behavior

If you do tempoup and then tempodown to normal speed - Kodi skips video frames (or moves to next keyframe maybe). Anyway it looks like it skips from 0.5 to 3 second of video.

## Possible Fix

this might be a continuation of https://github.com/xbmc/xbmc/issues/24324

### To Reproduce

Prerequisites:
1.
`.kodi/userdata/advancedsettings.xml`:
```
<?xml version="1.0" ?>
<advancedsettings>
    <video>
        <maxtempo>2.1</maxtempo>
    </video>
    <loglevel hide="false">1</loglevel>
</advancedsettings>
```
2.
`.kodi/userdata/keymaps/gen.xml`:
```
<?xml version="1.0"?>
<keymap>
  <fullscreenvideo>
    <keyboard>
      <key id="61568">playercontrol(tempoup)</key>
      <key id="61569">playercontrol(tempodown)</key>
    </keyboard>
  </fullscreenvideo>
</keymap>
```
3. Settings --> Video --> "Sync Playback to Display" option is enabled.

1. open video file
2. do tempoup few times
3. do tempodown same amount of times - so speed is back to 1
4. observe "skip" of video frames (or of few seconds of video)

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/getogetire

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Ubuntu 24.04.1 LTS, Android TV 10
 - Kodi version: 21.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by antonsoroko** [comment]
Looks like similar issue was also described in 2021: https://github.com/xbmc/xbmc/issues/19439

**Comment by kasimos** [comment]
yes, this issue is still the same as my old https://github.com/xbmc/xbmc/issues/19439 and I also tried it out again on Kodi 21.2 and the skipping is still present.

**Comment by mmortal03** [comment]
> yes, this issue is still the same as my old [#19439](https://github.com/xbmc/xbmc/issues/19439) and I also tried it out again on Kodi 21.2 and the skipping is still present.

I guess it's fine to let your old report become stale and close if someone will work to fix this one -- but it's weird that no one has picked any of these up to fix, given that the bug is at least easily reproducible.

**Comment by mmortal03** [comment]
@kasimos Please test the most recent v22 nightly. It seems that a recent pull request may have fixed this. ( https://github.com/xbmc/xbmc/pull/27644 )

**Comment by kasimos** [comment]
@mmortal03 yes, I can confirm with the highly (kodi-20260107-6ae18c73-master-41356) that the tempo up/down and back to normal speed is now working as intended.
There is no longer a weird skipped frame or laggy audio.
Everything is smooth now.
Thanks a lot!

**Comment by enen92** [comment]
Fixed by https://github.com/xbmc/xbmc/pull/27644
